state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1304,11 +1304,17 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	///
 	/// Blacklisted assets are not sent and will remain in storage.
 	fn do_egress_scheduled_ccm() {
+		let mut maybe_no_of_transfers_remaining =
+			T::FetchesTransfersLimitProvider::maybe_transfers_limit();
+
 		let ccms_to_send: Vec<CrossChainMessage<T::TargetChain>> =
 			ScheduledEgressCcm::<T, I>::mutate(|ccms: &mut Vec<_>| {
 				// Filter out disabled assets, and take up to batch_size requests to be sent.
-				ccms.extract_if(|ccm| !DisabledEgressAssets::<T, I>::contains_key(ccm.asset()))
-					.collect()
+				ccms.extract_if(|ccm| {
+					!DisabledEgressAssets::<T, I>::contains_key(ccm.asset()) &&
+						Self::should_fetch_or_transfer(&mut maybe_no_of_transfers_remaining)
+				})
```

Comment:
we can do better here by utilizing one more nonce since at this point we dont have to reserve one nonce for fetches. But I think we can leave it as is and leave this optimization (and potentially other optimizations) for when we actually do start hitting limits of parallel sol transactions, that is, we actually do have that much tx volume.

Reply:
yes, technically I believe we could use `maybe_fetches_limit` and check if it's > 0 to get one extra nonce but it's very strange. I agree we can leave it as a future optimization.

Reply:
we can do it in a better way by having a new function on the FetchesTransfersLimitProvider trait which would not affect the general code in ingress egress

Reply:
Added a ticket as an optimisation: https://linear.app/chainflip/issue/PRO-1497/solana-optimize-nonce-consumption